### PR TITLE
lift-db-reconciliation

### PR DESCRIPTION
**Pull request by Soumyadeep-Oracle**

### CODE REVIEW

**Review by AlexanderHodicke — CHANGES_REQUESTED** [review]
@oheimburger Code review please

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:1]: File name should be in lower case!

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+from sys.v_$INSTANCE i;
+PROMPT
+PROMPT
+SET PAGESIZE 0;
+SET HEADING OFF;
+SET VERIFY OFF;
+SET ECHO OFF;
+SET PAGES 0;
+
+ACCEPT user_response PROMPT 'Please press Enter to continue : '
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:41]: No space before colon (':'). It should read "continue: ".

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+--Source check and validation
+SET TERM OFF;
+SPOOL raise_exit.sql;
+DECLARE
+	v_mode varchar2(20);
+	c INT;
+BEGIN
+	select ltrim(open_mode) into v_mode from v$database;
+	IF (v_mode<>'READ WRITE') THEN
+		dbms_output.put_line('PROMPT Failed...Source Database is in '||v_mode||' Mode.');
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:52]: Space after ellipsis like "failed ... Source database is in".

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+		dbms_output.put_line('exit;');	
+	ELSE
+	    dbms_output.put_line('PROMPT Passed...Source Database is in '||v_mode||' Mode.');
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:55]: See comment for line 52.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+		dbms_output.put_line('PROMPT Passed...Source Checks Passed...');
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:56]: See comment for line 52.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+    commit;	
+	END;
+	dbms_output.put_line('PROMPT Passed...Temporary user list generated...');
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:67]: See comment for line 52.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 6]
```diff
@@ -0,0 +1,236 @@
...
+	
+END;
+/
+SPOOL OFF
+SET TERMOUT ON;
+SPOOL Source_Steps_Execution.log append
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:73]: File name should be in lower case.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 7]
```diff
@@ -0,0 +1,236 @@
...
+PROMPT
+PROMPT OOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOOO
+PROMPT
+DECLARE	
+c INT;
+BEGIN
+	SELECT COUNT(1) INTO c FROM USER_TABLES WHERE table_name = 'USER_TEMP';
+	IF c = 1 THEN
+		EXECUTE immediate 'DROP TABLE USER_TEMP PURGE';
+		dbms_output.put_line('Passed...Source Steps compleated Successfully!!!');
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:226]: See comment on line 52.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql [hunk 8]
```diff
@@ -0,0 +1,236 @@
...
+	END IF;
+END;
+/
+PROMPT
+SET TERM ON;
+SPOOL Source_Steps_Execution.log append;
+PROMPT Passed...Source Steps compleated Successfully!!!
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:233]: See comment on line 52.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Target.sh [hunk 9]
```diff
@@ -0,0 +1,1641 @@
+#!/bin/bash
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v1.0 as shown at https://oss.oracle.com/licenses/upl.
+# Purpose: Script to capture Target Object count,table row count and invalid objects and generate reconcile report.
+# The purpose of this script is to verify the migrated data against the original source.
+# @author: Ananda Ghosh Dastidar.
+
+if [ "$#" -lt 2 ]; then
+   echo Usage: $0 "o2o/o2o@service_name" "spool_dir"
```
Comment (oheimburger) [inline cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Target.sh:9]: No user credentials!

**Review by oheimburger — CHANGES_REQUESTED** [review]
THIS BRANCH CONTAINS USER CREDENTIALS AND SHOULD BE DELETED!

Apply all the requested changes to your local copy and commit a new branch.

**Review by oheimburger — CHANGES_REQUESTED** [review]
Please resolve pending changes.

**Comment by AlexanderHodicke** [comment]
@Soumyadeep-Oracle please fix the feedback from Olaf or close this PR. thanks

**Review by oheimburger — CHANGES_REQUESTED** [review]
Please resolve my comments.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:1]: Resolved

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql (hunk 1 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:41]: Resolved

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql (hunk 2 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:52]: This seems to be fine.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql (hunk 3 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:55]: This seems to be fine.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql (hunk 5 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:67]: This seems to be fine.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql (hunk 4 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:56]: This seems to be fine.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql (hunk 6 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:73]: Resolved.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql (hunk 7 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Source.sql:226]: This seems to be fine.

File: cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Target.sh (hunk 9 above)
Comment (Soumyadeep-Oracle) [reply cloud-architecture/custom-apps-and-consolidation/db-reconciliation/files/MF_DP_Recon_Target.sh:9]: Resolved.

**Review by oheimburger — APPROVED** [review]

**Comment by oheimburger** [comment]
Branch is outdated. Please update the branch.

**Comment by oheimburger** [comment]
@AlexanderHodicke Please delete this branch. I can't do it.

**Pull request closed without merging**